src/lang-go.ts

```diff
@@ -563,6 +558,121 @@ function positionParams(doc: sourcegraph.TextDocument, pos: sourcegraph.Position
+    sourcegraph.app.activeWindow!.showNotification(
+        [
+            '**❌ Unable to access to the Go language server**',
+            '',
+            'Your browser could not access:',
+            '',
+            '```',
+            `"go.serverUrl": "${address}"`,
+            '```',
+            '',
+            ...(['ws:', 'wss:'].includes(address.protocol)
+                ? [
+                      'Troubleshoot using these commands:',
+                      '',
+                      `- **\`ping ${address.hostname}\`**`,
+                      `- **\`telnet ${address.hostname} ${address.port || portByProtocol[address.protocol]}\`**`,
+                      '',
+                  ]
+                : [
+                      `The protocol **\`${
+                          address.protocol
+                      }\`** is invalid. Only **\`ws:\`** or **\`wss:\`** are valid.`,
+                      '',
+                  ]),
+            '',
+            `To apply the fix:`,
+            '',
+            `- Correct the value in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+            '',
+            'To temporarily disable:',
+            '',
+            `- Comment out **\`go.serverUrl\`** in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+        ].join('\n')
+    )
+}
+
+function notifyLanguageServerUnableToConnectToSourcegraph(): void {
+    sourcegraph.internal.updateContext({
+        showError: true,
+        'codeIntel.error.message': 'Error in language server setup. Click to open documentation.',
+        'codeIntel.error.link': 'https://sourcegraph.com/extensions/sourcegraph/go',
+    })
+    sourcegraph.app.activeWindow!.showNotification(
+        [
+            '**❌ Error in Go language server setup**',
+            '',
+            `The Go language server was unable to fetch repository contents from Sourcegraph at:`,
+            '',
+            '```',
+            `"go.sourcegraphUrl": "${sourcegraphURL()}"`,
+            '```',
+            '',
+            'To troubleshoot this, **`docker exec`** into the go-langserver container and run:',
+            '',
+            `- **\`ping ${sourcegraphURL().hostname}\`**`,
+            `- **\`curl ${sourcegraphURL()}\`**`,
+            '',
+            `To apply the fix:`,
+            '',
+            `- Correct the value in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+            '',
+            'To temporarily disable:',
+            '',
+            `- Comment out **\`go.serverUrl\`** in your [global settings](${
+                sourcegraph.internal.sourcegraphURL
+            }site-admin/global-settings).`,
+        ].join('\n')
+    )
+}
+
+async function canary(address: URL): Promise<void> {
```

Comment:
Could you add a doc comment? It's not obvious to me what this function's job is (maybe I just don't understand the name)